src/plugins/intel_gpu/src/graph/impls/common/loop.cpp

```diff
@@ -111,6 +111,10 @@ struct loop_impl : typed_primitive_impl<loop> {
     }
 
     event::ptr execute_impl(const std::vector<event::ptr>& events, loop_inst& instance) override {
+        // Wait for event completion
+        for (auto& e : events) {
+            e->wait();
+        }
```

Comment:
I believe this will cause regression.. 
I guess that the target issue is the event of "execution condition" right? 
then I think we need to wait for only that event specifically, not for all events..

Reply:
I updated the event wait for only execution_condition. Thanks!

Reply:
@yeonbok No performance regression in several loop including models. Thanks!